src/Makefile

```diff
@@ -929,6 +929,9 @@ net: netvariables
 		fi; \
 	fi; \
 
+format:
+	find . -type f \( -iname "*.h" -o -iname "*.cpp" \) ! -name "incbin.h" | xargs clang-format -i -style=file
```

Comment:
don't like the find so much here, I would prefer if we could do it from the list of objects somehow. It might not be easy with the .h files, however.

Reply:
I thought about this too, since we already hardcode all  `SRCS` files anyway, I could probably just add all the missing headers to that... 
This would be also more portable, if you are using windows it tries to use the windows [find](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/find), instead of using the find in msys, even if that is in the users environment path

Reply:
not only, that, but if you have some none-git file the above would pick it up as well.

it might be possible to get make to list all dependencies of the object files, which would be a list of all .cpp and .h (and maybe some stuff that needs to be filtered out).

Reply:
You already (kinda) have that information in the `.depend` file. I'll have a look if theres some nice way to get this just for main without having to compile
```
main.o: main.cpp bitboard.h types.h tune.h evaluate.h misc.h position.h \
 nnue/nnue_accumulator.h nnue/nnue_architecture.h \
 nnue/features/half_ka_v2_hm.h nnue/features/../../misc.h \
 nnue/features/../../types.h nnue/features/../../tune.h \
 nnue/features/../nnue_common.h nnue/features/../../misc.h \
 nnue/layers/affine_transform.h nnue/layers/../nnue_common.h \
 nnue/layers/simd.h nnue/layers/affine_transform_sparse_input.h \
 nnue/layers/../../bitboard.h nnue/layers/affine_transform.h \
 nnue/layers/clipped_relu.h nnue/layers/sqr_clipped_relu.h \
 nnue/nnue_common.h search.h movepick.h movegen.h thread.h \
 thread_win32_osx.h uci.h
```

Reply:
let me have a look, I think this can be done 'easily'

Reply:
```
format: .depend
	sed ':a;/\\$$/{N;s/\\\n//;ba}' .depend | cut -d':' -f2 | sed "s/ /\n/g" | xargs -n1 realpath | sort | uniq | grep -v incbin | xargs clang-format -i -style=file
```

Reply:
not so sure how portable this is, realpath is from the gnu core-utils, and we need `xargs -n1 realpath | sort | uniq` to have all files just once.

Reply:
This requires that there is a .depend present no? Which is maybe outdated or not there?

Reply:
Even with the msys2 shell on windows this fails for me :/ 
I think the most portable way would be to hardcode all files to format...

Reply:
also an option, not too ugly.

Reply:
we could have a separate batch script for this that's only called from the makefile? to make it a little bit less daunting

Reply:
`awk '{for(i=1;i<=NF;i++) print $i}' .depend  | grep -v '[\:]$' | xargs realpath | sort | uniq` this also works from the cli. (in the makefile needs proper escaping).